### PR TITLE
[lodash] Add uniqWith to array wrappers interfaces

### PR DESCRIPTION
**Pull request by ivanvoznyakovsky**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://lodash.com/docs/4.17.4#uniqWith

### CODE REVIEW

File: lodash/lodash-tests.ts [hunk 1]
```diff
@@ -2245,6 +2245,46 @@ namespace TestUniqBy {
...
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').chain().uniqWith<string[]>(comparator);
```
Comment (zuzusik) [inline lodash/lodash-tests.ts]: I believe tests shouldn't enforce typings (< type > enforces typing, not specifies it, look [at this example ](http://www.typescriptlang.org/play/#src=type%20TMyType%20%3D%20%7B%20b%3A%20string%20%7D%3B%0Afunction%20test(t%3A%20TMyType)%20%7B%20%7D%3B%0Aconst%20a%20%3D%20(%3CTMyType%3E%20%7B%7D)%3B%0Aconst%20b%20%3D%20%7B%7D%0Atest(a)%3B%0Atest(b)%3B%0A))

Comment (minestarks) [inline lodash/lodash-tests.ts]: I believe you're referring to type assertions, but these are [type arguments](https://www.typescriptlang.org/docs/handbook/generics.html).

**Comment by minestarks** [comment]
@ivanvoznyakovsky Could you please fix the failing test before I can merge? Thank you.

File: lodash/lodash-tests.ts (hunk 1 above)
Comment (zuzusik) [inline lodash/lodash-tests.ts]: This is type arguments but we enforce it. 

Ideally TS bases on Typings should be able to understand true that the result of `_(<string>).chain().uniqWith(comparator)` is `_.LoDashImplicitArrayWrapper<string[]>` 

Is TS typings system capable of handling such cases? If it is then probably it is required to improve typings of `chain` or `_`.

**Comment by ivanvoznyakovsky** [comment]
@minestarks not sure I see how my changes are failing tests.

> Error in lodash.meanby
Must specify "lib", usually to `"lib": ["es6"]` or `"lib": ["es6", "dom"]`.

**Comment by zuzusik** [comment]
Maybe you just need to rebase over the latest master?

**Comment by ivanvoznyakovsky** [comment]
yeah. that could be the reason. but after rebase it's still the same error.

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by paulvanbrenk** [comment]
Closing this, please re-open when you've resolved the open issues.

**Pull request closed without merging**